From cac00c154be5c8941d40b972bdcc0eb8bf209b93 Mon Sep 17 00:00:00 2001
From: Christian Stewart <christian@paral.in>
Date: Sun, 2 May 2021 19:04:50 -0700
Subject: [PATCH] install_helper: fix sysconf dir for init.d script

Fixes the following build error in Buildroot as a host package:

Running custom install script 'install_helper.sh /host/etc /host/bin /host/lib/udev/rules.d false
+ sysconfdir=/host/etc
+ bindir=/host/bin
+ udevrulesdir=/host/lib/udev/rules.d
+ useroot=false
+ '[' -z '' ']'
+ DESTDIR=
+ install -D -m 644 /build/host-libfuse3-3.10.3/util/fuse.conf /host/etc/fuse.conf
+ false
+ install -D -m 644 /build/host-libfuse3-3.10.3/util/udev.rules /host/lib/udev/rules.d/99-fuse3.rules
+ install -D -m 755 /build/host-libfuse3-3.10.3/util/init_script /etc/init.d/fuse3
install: cannot create regular file '/etc/init.d/fuse3': Permission denied
FAILED: install script 'install_helper.sh /host/etc /host/bin /host/lib/udev/rules.d false' exit code 1, stopped
FAILED: meson-install

Signed-off-by: Christian Stewart <christian@paral.in>
---
 util/install_helper.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/util/install_helper.sh b/util/install_helper.sh
index cb649a7..f4a6163 100755
--- a/util/install_helper.sh
+++ b/util/install_helper.sh
@@ -40,14 +40,14 @@ install -D -m 644 "${MESON_SOURCE_ROOT}/util/udev.rules" \
         "${DESTDIR}${udevrulesdir}/99-fuse3.rules"
 
 install -D -m 755 "${MESON_SOURCE_ROOT}/util/init_script" \
-        "${DESTDIR}/etc/init.d/fuse3"
+        "${DESTDIR}${sysconfdir}/init.d/fuse3"
 
 
 if test -x /usr/sbin/update-rc.d && test -z "${DESTDIR}"; then
     /usr/sbin/update-rc.d fuse3 start 34 S . start 41 0 6 . || /bin/true
 else
     echo "== FURTHER ACTION REQUIRED =="
-    echo "Make sure that your init system will start the ${DESTDIR}/etc/init.d/fuse3 init script"
+    echo "Make sure that your init system will start the ${DESTDIR}${sysconfdir}/init.d/fuse3 init script"
 fi
 
 
-- 
2.31.1

